cpp/benchmarks/ast/transform.cpp

```diff
@@ -86,38 +85,23 @@ static void BM_ast_transform(benchmark::State& state)
 
   auto const& expression_tree_root = expressions.back();
 
-  // Execute benchmark
-  for (auto _ : state) {
-    cuda_event_timer raii(state, true);  // flush_l2_cache = true, stream = 0
-    cudf::compute_column(table, expression_tree_root);
-  }
+  state.exec(nvbench::exec_tag::sync,
+             [&](nvbench::launch&) { cudf::compute_column(table, expression_tree_root); });
 
   // Use the number of bytes read from global memory
-  state.SetBytesProcessed(static_cast<int64_t>(state.iterations()) * state.range(0) *
-                          (tree_levels + 1) * sizeof(key_type));
-}
-
-static void CustomRanges(benchmark::internal::Benchmark* b)
-{
-  auto row_counts       = std::vector<cudf::size_type>{100'000, 1'000'000, 10'000'000, 100'000'000};
-  auto operation_counts = std::vector<cudf::size_type>{1, 5, 10};
-  for (auto const& row_count : row_counts) {
-    for (auto const& operation_count : operation_counts) {
-      b->Args({row_count, operation_count});
-    }
-  }
+  state.add_global_memory_reads(static_cast<int64_t>(state.get_summaries().size()) * table_size *
+                                (tree_levels + 1) * sizeof(key_type));
```

Comment:
```suggestion
  state.add_global_memory_reads<key_type>(table_size * (tree_levels + 1));
```
Nvbench and gbench compute memory bandwidth in a slightly different way, so there's no need to manually calculate the number of bytes. After making this change, could you please double-check to ensure that the gbench results align coherently with the nvbench ones?

for your reference: https://github.com/NVIDIA/nvbench/blob/e8c8877d3669e30e977cf6258d96e1c645922077/examples/throughput.cu